code/modules/mob/living/emote.dm

```diff
@@ -218,7 +218,8 @@
 /datum/emote/living/laugh/get_sound(mob/living/user)
 	if(ishuman(user))
 		var/mob/living/carbon/human/H = user
-		if(H.dna.species.id == "human" && (!H.mind || !H.mind.miming))
+		var/human_laugh = ishumanbasic(H) || isfelinid(H)
+		if(human_laugh && (!H.mind || !H.mind.miming))
```

Comment:
ishumanbasic() returns TRUE for felinids already,
also, this code is bad and snowflakey.

add a `var/list/laughs` to `/datum/species`. then have something like
`if(H.dna.species.laughs)`

Reply:
I've done as you suggested and it doesn't seem to work, am I supposed to assign something to the list?

Reply:
> 

You're supposed to populate the list with file paths to each laugh, then you can modify the code to `pick()` from the species laugh list